src/components/UserDashboard/AuthorizedStudies/index.js

```diff
@@ -1,112 +1,100 @@
 import React, { Fragment } from 'react';
-import { compose } from 'recompose';
+import { compose, lifecycle } from 'recompose';
+import { Link } from 'react-router-dom';
 import { injectState } from 'freactal';
 import { withTheme } from 'emotion-theming';
-import { connectGen3, getAccessToken } from 'services/gen3';
-import { Gen3UserProvider } from 'services/gen3';
+import { Trans } from 'react-i18next';
+import Component from 'react-component-component';
+import { isEmpty } from 'lodash';
+
 import CardHeader from 'uikit/Card/CardHeader';
 import DownloadController from 'icons/DownloadController';
 
-import Component from 'react-component-component';
+import { provideFenceConnections } from 'stateProviders';
 import { withApi } from 'services/api';
-import { GEN3 } from 'common/constants';
-import Gen3Connected from './Gen3Connected';
+import StudiesConnected from './StudiesConnected';
 
 import AccessGate from '../../AccessGate';
 import { DashboardCard, CardContentSpinner } from '../styles';
 
-import {
-  trackUserInteraction,
-  analyticsTrigger,
-  TRACKING_EVENTS,
-} from 'services/analyticsTracking';
-
 import Info from '../Info';
 
-import { ConnectButton } from '../styles';
+import { CardActionButton } from '../styles';
 
 const AuthorizedStudies = compose(
   withApi,
+  provideFenceConnections,
   injectState,
   withTheme,
-)(({ state: { integrationTokens, loggedInUser }, effects, theme, api, ...props }) => (
-  <Component initialState={{ connected: false, badgeNumber: null }}>
-    {({ setState, state }) => {
-      const Header = <CardHeader title="Authorized Studies" badge={state.badgeNumber} />;
+  lifecycle({
+    componentDidMount() {
+      const { api } = this.props;
+      this.props.effects.fetchFenceConnections({ api });
+    },
+  }),
```

Comment:
would it make sense to move this to the `<Component />` below with `didMount`? Just because these big chain of HOCs can get confusing about where things come from, like `this.props.effects.fetchFenceConnections` :P naming `didMount` can help avoid the excessive inline lambdas :)

Reply:
Can do. No reason to mix usage like I did.

Reply:
👍